File: src/main/java/io/bloco/faker/components/Business.java
package io.bloco.faker.components;

import org.joda.time.DateTime;

import java.util.Date;

import io.bloco.faker.FakerComponent;
import io.bloco.faker.FakerData;

public class Business extends FakerComponent {

    private static final int CREDIT_CARD_PLUS_YEARS_MAX = 4;

    public Business(FakerData data) {
        super(data);
    }

    public String creditCardNumber() {
        return fetch("business.credit_card_numbers");
    }

    public Date creditCardExpireDate() {
        return DateTime.now()
                .plusYears(randomHelper.number(CREDIT_CARD_PLUS_YEARS_MAX) + 1)
                .toDate();
    }

    public String creditCardType() {
        return fetch("business.credit_card_types");
    }
}

File: lib/src/main/kotlin/io/bloco/faker/components/Business.kt
package io.bloco.faker.components

import io.bloco.faker.FakerComponent
import io.bloco.faker.FakerData
import io.bloco.faker.helpers.RandomHelper
import java.time.LocalDate

class Business(data: FakerData) : FakerComponent(data) {
    fun creditCardNumber(): String {
        return fetch("business.credit_card_numbers")
    }

    fun creditCardExpireDate(): LocalDate {
        return LocalDate.now()
            .plusYears(RandomHelper.number(CREDIT_CARD_PLUS_YEARS_MAX).toLong() + 1)
    }

    fun creditCardType(): String {
        return fetch("business.credit_card_types")
    }

    companion object {
        private const val CREDIT_CARD_PLUS_YEARS_MAX = 4
    }
}
